meta/store_test.go

```diff
@@ -389,6 +389,40 @@ func TestStore_CreateShardGroup(t *testing.T) {
 	} else if sgi.ID != 1 {
 		t.Fatalf("unexpected shard group: %#v", sgi)
 	}
+
+	// Check that it is returned correctly when requested.
+	if sgs, err := s.ShardGroups("db0", "rp0"); err != nil {
+		t.Fatal(err)
+	} else if len(sgs) != 1 {
+		t.Fatalf("unexpected number of shard groups returned: %d", len(sgs))
+	} else if sgs[0].ID != 1 {
+		t.Fatalf("unexpected shard group: %#v", sgs[0])
+	}
+
+	if sgs, err := s.ShardGroupsByTimeRange("db0", "rp0", time.Date(1999, time.January, 1, 0, 0, 0, 0, time.UTC), time.Date(1999, time.January, 2, 0, 0, 0, 0, time.UTC)); err != nil {
+		t.Fatal(err)
+	} else if len(sgs) != 0 {
+		t.Fatalf("unexpected number of shard groups returned: %d", len(sgs))
+	}
+	if sgs, err := s.ShardGroupsByTimeRange("db0", "rp0", time.Date(2000, time.January, 1, 0, 0, 0, 0, time.UTC), time.Date(2001, time.January, 1, 0, 0, 0, 0, time.UTC)); err != nil {
+		t.Fatal(err)
+	} else if len(sgs) != 1 {
+		t.Fatalf("unexpected number of shard groups returned: %d", len(sgs))
+	} else if sgs[0].ID != 1 {
+		t.Fatalf("unexpected shard group: %#v", sgs[0])
+	}
+	if sgs, err := s.ShardGroupsByTimeRange("db0", "rp0", time.Date(1999, time.January, 1, 0, 0, 0, 0, time.UTC), time.Date(2001, time.January, 1, 0, 0, 0, 0, time.UTC)); err != nil {
+		t.Fatal(err)
+	} else if len(sgs) != 1 {
+		t.Fatalf("unexpected number of shard groups returned: %d", len(sgs))
+	} else if sgs[0].ID != 1 {
+		t.Fatalf("unexpected shard group: %#v", sgs[0])
+	}
+	if sgs, err := s.ShardGroupsByTimeRange("db0", "rp0", time.Date(2002, time.January, 1, 0, 0, 0, 0, time.UTC), time.Date(2002, time.January, 2, 0, 0, 0, 0, time.UTC)); err != nil {
+		t.Fatal(err)
+	} else if len(sgs) != 0 {
+		t.Fatalf("unexpected number of shard groups returned: %d", len(sgs))
+	}
 }
```

Comment:
Well, let me ask you this. What is a clean way to test it, without, in effect, just retesting initial shard creation? There really isn't a way that I can see. That why I kinda just slammed it in here. ;-)